kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -172,32 +176,44 @@ public Snapshot getSnapshotAt(Engine engine, long version) throws TableNotFoundE
    * @return a {@link Snapshot} of the table at the provided timestamp
    * @throws TableNotFoundException
    */
-  public Snapshot getSnapshotForTimestamp(Engine engine, long millisSinceEpochUTC)
+  public Snapshot getSnapshotForTimestamp(
+      Engine engine, long millisSinceEpochUTC, SnapshotQueryContext snapshotContext)
       throws TableNotFoundException {
     long startTimeMillis = System.currentTimeMillis();
     long versionToRead =
-        DeltaHistoryManager.getActiveCommitAtTimestamp(
-                engine,
-                logPath,
-                millisSinceEpochUTC,
-                true /* mustBeRecreatable */,
-                false /* canReturnLastCommit */,
-                false /* canReturnEarliestCommit */)
-            .getVersion();
+        snapshotContext
+            .getSnapshotMetrics()
+            .timestampToVersionResolutionDuration
+            .time(
+                () ->
+                    DeltaHistoryManager.getActiveCommitAtTimestamp(
+                            engine,
+                            logPath,
+                            millisSinceEpochUTC,
+                            true /* mustBeRecreatable */,
+                            false /* canReturnLastCommit */,
+                            false /* canReturnEarliestCommit */)
+                        .getVersion());
     logger.info(
         "{}: Took {}ms to fetch version at timestamp {}",
         tablePath,
         System.currentTimeMillis() - startTimeMillis,
```

Comment:
Instead of creating the local variable `startTimeMillis` could we just use the `timestampToVersionResolutionDuration` that we just set? Only if that would improve the cleanliness and reduce the amount of code ...

Reply:
I updated it to do this